monix-execution/shared/src/main/scala/monix/execution/internal/exceptions.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2014-2021 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.execution.internal
+
+private[monix] object exceptions {
+  /** 
+    * To use in `case _ =>` branches that are absurd, but needed
+    * due to the compiler complaining.
+    */  
+  def matchError[A](received: A): Nothing =
```

Comment:
This was needed because Scala 2.13.5 now complains on missing cases when using the partial-function syntax, when a `Function1` is requested as parameter. Pretty cool, not sure why it wasn't working before.

But we now have to throw MatchErrors manually in some cases.